andela/boondocks-bn-backend
# #169257390 Integrate Coveralls test coverage

#### What does this PR do?
Implements test coverage on the codebase using Coveralls.io

#### Description of Task to be completed?
- [x] install the following dependencies:
   - [x] mocha
   - [x] coveralls
   - [x] nyc
- [x] add further configuration to the 'travis.yml' file to run the 'test' and 'coverage' scripts using 'nyc'
- [x] update the README file with the Coveralls badge
- [x] update '.gitignore' not to track coverage files generated by 'nyc', namely '.nyc_output' 

#### How should this be manually tested?
- After cloning the repo 'cd' into it
- run `npm install` to install all the dependencies
- create a branch from this branch, add some tests to './src/tests'
- run the tests locally by navigating to the project folder using Terminal and running `npm test`
- your tests will either pass or fail and you should see a coverage report printed out on Terminal
- push your branch to Github to trigger a TravisCI build
- after the tests have run you can get on Github, under branches you will be able to see the checks that have been performed on your branch as having passed or failed as well as links to TravisCI and Coveralls respectively. View the coverage report on Coveralls.io 
- once you merge your branch to develop, the README file will refresh after a couple of hours (about 12 hours) to reflect the latest coverage on the newly updated develop branch

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#169257390](https://www.pivotaltracker.com/story/show/169257390)

#### Screenshots (if appropriate)
<img width="1440" alt="coveralls running" src="https://user-images.githubusercontent.com/17108099/68232811-92bec100-0006-11ea-9227-4dc4bd0ce340.png">
<img width="1035" alt="README with Coveralls badge" src="https://user-images.githubusercontent.com/17108099/68235067-35793e80-000b-11ea-956c-1edda9a0cb3d.png">

#### Questions:
N/A
